app/javascript/add-another.js

```diff
@@ -105,6 +110,38 @@ const showItemRemovedFeedback = (section) => {
   button.addEventListener("click", () => { feedback.remove() });
 };
 
+const updateErrorMessages = (sectionToRemove, sectionList) => {
+  let reachedSectionToRemove = false
+  sectionList.querySelectorAll('[data-add-another-role="section"]').forEach((currentSection, index) => {
+    const currentSectionCurrentPosition = index + 1;
+    // Remove all error summary messages pertaining to an item that has been removed
+    if (currentSection === sectionToRemove) {
+      reachedSectionToRemove = true;
+      document.querySelectorAll(`[data-add-another-role="errorMessage"][data-add-another-item-position="${currentSectionCurrentPosition}"]`).forEach((redundantErrorMessage) => {
+        redundantErrorMessage.closest("li").remove();
+      });
+    // Decrement the numbers of all error messages that are for items that come after the removed item.
+    } else if (reachedSectionToRemove) {
+      const newPosition = index;
+      document.querySelectorAll(`[data-add-another-role="errorMessage"][data-add-another-item-position="${currentSectionCurrentPosition}"]`).forEach((elementToUpdate) => {
+        elementToUpdate.dataset.addAnotherItemPosition = newPosition;
+        elementToUpdate.querySelectorAll('[data-add-another-role="errorPosition"]').forEach((positionText) => {
+          positionText.innerHTML = newPosition;
+        })
+
+        elementToUpdate.closest("a").href = elementToUpdate.closest("a").href.replace(`-${currentSectionCurrentPosition}-`, `-${newPosition}-`);
```

Comment:
ok - so once we've worked out which are the error messages after the one we've just removed, this line, is the _thing_ that replaces the current postion (`currentSectionCurrentPosition`) with the new position (`newPosition`).

Reply:
Yes, line 129 sets the visible text, line 132 updates the location of the clickable link